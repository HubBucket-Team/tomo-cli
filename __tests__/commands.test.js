import {
    getDirectoryTree,
    run,
    useTemporaryDirectory
} from './tomo-test';
import addEslint from '../src/commands/add-eslint';
import addMarionette from '../src/commands/add-marionette';
import addWebpack from '../src/commands/add-webpack';

jest.mock('execa');

describe('Commands', () => {
    let tempDirectory;
    const skipInstall = true;
    const [setTempDir, cleanupTempDir] = useTemporaryDirectory();
    beforeEach(async () => {
        tempDirectory = await setTempDir();
        process.chdir(tempDirectory);
    });
    afterEach(async () => {
        await cleanupTempDir();
    });
    xtest('add-eslint', async () => {
        const sourceDirectory = 'src';
        const options = {skipInstall, sourceDirectory};
        await run(addEslint, options);
        const tree = getDirectoryTree(tempDirectory, {omit: ['extension', 'path']});
        console.log(tree);
        // expect(tree).toMatchSnapshot();
    });
    xtest('add-marionette', async () => {
        const sourceDirectory = 'src';
        const options = {skipInstall, sourceDirectory};
        await run(addMarionette, options);
        const tree = getDirectoryTree(tempDirectory, {omit: ['extension', 'path']});
        expect(tree).toMatchSnapshot();
    });
    xtest('add-webpack', async () => {
        const sourceDirectory = 'src';
        const options = {skipInstall, sourceDirectory};
        await run(addWebpack, options);
        const tree = getDirectoryTree(tempDirectory, {omit: ['extension', 'path']});
        console.log(tree);
        // expect(tree).toMatchSnapshot();
    });
});
