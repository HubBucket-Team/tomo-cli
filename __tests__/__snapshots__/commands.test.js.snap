// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"Add" commands add-babel (with React) 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: babel.config.js,
            size: 318,
            type: file
        }
    ],
    size: 318,
    type: directory
}
"
`;

exports[`"Add" commands add-babel (with React) 2`] = `
"module.exports = {
    plugins: [
        '@babel/plugin-transform-runtime',
        '@babel/plugin-proposal-class-properties',
        '@babel/plugin-proposal-export-default-from',
        '@babel/plugin-proposal-optional-chaining'
    ],
    presets: [
        @babel/preset-env,
        @babel/preset-react
    ]
};"
`;

exports[`"Add" commands add-babel 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: babel.config.js,
            size: 291,
            type: file
        }
    ],
    size: 291,
    type: directory
}
"
`;

exports[`"Add" commands add-babel 2`] = `
"module.exports = {
    plugins: [
        '@babel/plugin-transform-runtime',
        '@babel/plugin-proposal-class-properties',
        '@babel/plugin-proposal-export-default-from',
        '@babel/plugin-proposal-optional-chaining'
    ],
    presets: [
        '@babel/preset-env'
    ]
};"
`;

exports[`"Add" commands add-esdoc (with React) 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: esdoc.conf.json,
            size: 417,
            type: file
        }
    ],
    size: 417,
    type: directory
}
"
`;

exports[`"Add" commands add-esdoc (with React) 2`] = `
"{
    \\"source\\": \\"./src\\",
    \\"destination\\": \\"./docs\\",
    \\"plugins\\": [
        {
            \\"name\\": \\"esdoc-standard-plugin\\"
        },
        {
            \\"name\\": \\"esdoc-ecmascript-proposal-plugin\\",
            \\"option\\": {
                \\"all\\": true
            }
        },
        {
            \\"name\\": \\"esdoc-jsx-plugin\\",
            \\"options\\": {
                \\"enable\\": true
            }
        }
    ]
}
"
`;

exports[`"Add" commands add-esdoc 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: esdoc.conf.json,
            size: 286,
            type: file
        }
    ],
    size: 286,
    type: directory
}
"
`;

exports[`"Add" commands add-esdoc 2`] = `
"{
    \\"source\\": \\"./src\\",
    \\"destination\\": \\"./docs\\",
    \\"plugins\\": [
        {
            \\"name\\": \\"esdoc-standard-plugin\\"
        },
        {
            \\"name\\": \\"esdoc-ecmascript-proposal-plugin\\",
            \\"option\\": {
                \\"all\\": true
            }
        }
    ]
}
"
`;

exports[`"Add" commands add-eslint (with React) 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 353,
            type: file
        }
    ],
    size: 389,
    type: directory
}
"
`;

exports[`"Add" commands add-eslint (with React) 2`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        omaha-prime-grade,
        plugin:react/recommended
    ],
    parser: 'babel-eslint',
    parserOptions: {
        ecmaFeatures: {
            jsx: true
        }
    },
    settings: {
        react: {
            version: 'undefined'
        }
    }
};"
`;

exports[`"Add" commands add-eslint 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 154,
            type: file
        }
    ],
    size: 190,
    type: directory
}
"
`;

exports[`"Add" commands add-eslint 2`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint'
};"
`;

exports[`"Add" commands add-jest 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": []
}
"
`;

exports[`"Add" commands add-jest 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:watch\\": \\"npm test -- --watchAll\\"
    }
}
"
`;

exports[`"Add" commands add-marionette 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: assets,
            children: [
                {
                    name: css,
                    children: [
                        {
                            name: style.css,
                            size: 38,
                            type: file
                        }
                    ],
                    size: 38,
                    type: directory
                },
                {
                    name: templates,
                    children: [
                        {
                            name: example.hbs,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                }
            ],
            size: 38,
            type: directory
        },
        {
            name: src,
            children: [
                {
                    name: app.js,
                    size: 435,
                    type: file
                },
                {
                    name: index.html,
                    size: 531,
                    type: file
                },
                {
                    name: main.js,
                    size: 1080,
                    type: file
                },
                {
                    name: plugins,
                    children: [
                        {
                            name: mn.radio.logging.js,
                            size: 3807,
                            type: file
                        },
                        {
                            name: mn.redux.state.js,
                            size: 2184,
                            type: file
                        }
                    ],
                    size: 5991,
                    type: directory
                },
                {
                    name: shims,
                    children: [
                        {
                            name: mn.renderer.shim.js,
                            size: 486,
                            type: file
                        },
                        {
                            name: mn.templates.shim.js,
                            size: 662,
                            type: file
                        }
                    ],
                    size: 1148,
                    type: directory
                }
            ],
            size: 9185,
            type: directory
        }
    ],
    size: 9223,
    type: directory
}
"
`;

exports[`"Add" commands add-postcss 1`] = `
"module.exports = {
    parser: require('postcss-safe-parser'),
    processors: [
        require('stylelint')(),
        require('postcss-import')(),
        require('postcss-cssnext')(),
        require('uncss').postcssPlugin({html: ['index.html']}),
        require('cssnano')(),
        require('postcss-reporter')({clearReportedMessages: true})
    ]
};"
`;

exports[`"Add" commands add-postcss 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": []
}
"
`;

exports[`"Add" commands add-webpack 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: babel.config.js,
            size: 318,
            type: file
        },
        {
            name: webpack.config.js,
            size: 650,
            type: file
        }
    ],
    size: 968,
    type: directory
}
"
`;

exports[`"Add" commands add-webpack 2`] = `
"const {resolve} = require('path');
const DashboardPlugin = require('webpack-dashboard/plugin');

module.exports = {
    mode: 'development',
    entry: {
        app: 'src/main.js'
    },
    output: {
        path: resolve('./dist'),
        filename: 'bundle.min.js'
    },
    module: {
        rules: [
            {
                test: /.js?$/,
                exclude: /node_modules/,
                loader: 'babel-loader',
                query: {
                    presets: [
                        '@babel/env'
                    ]
                }
            }
        ]
    },
    plugins: [
        new DashboardPlugin()
    ]
};"
`;

exports[`"Create/New" commands create new app 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 154,
            type: file
        },
        {
            name: assets,
            children: [
                {
                    name: css,
                    children: [
                        {
                            name: style.css,
                            size: 38,
                            type: file
                        }
                    ],
                    size: 38,
                    type: directory
                },
                {
                    name: templates,
                    children: [
                        {
                            name: example.hbs,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                }
            ],
            size: 38,
            type: directory
        },
        {
            name: babel.config.js,
            size: 291,
            type: file
        },
        {
            name: package.json,
            size: 473,
            type: file
        },
        {
            name: src,
            children: [
                {
                    name: app.js,
                    size: 435,
                    type: file
                },
                {
                    name: index.html,
                    size: 531,
                    type: file
                },
                {
                    name: main.js,
                    size: 1080,
                    type: file
                },
                {
                    name: plugins,
                    children: [
                        {
                            name: mn.radio.logging.js,
                            size: 3807,
                            type: file
                        },
                        {
                            name: mn.redux.state.js,
                            size: 2184,
                            type: file
                        }
                    ],
                    size: 5991,
                    type: directory
                },
                {
                    name: shims,
                    children: [
                        {
                            name: mn.renderer.shim.js,
                            size: 486,
                            type: file
                        },
                        {
                            name: mn.templates.shim.js,
                            size: 662,
                            type: file
                        }
                    ],
                    size: 1148,
                    type: directory
                }
            ],
            size: 9185,
            type: directory
        }
    ],
    size: 10177,
    type: directory
}
"
`;

exports[`"Create/New" commands create new project 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 154,
            type: file
        },
        {
            name: babel.config.js,
            size: 291,
            type: file
        },
        {
            name: package.json,
            size: 473,
            type: file
        },
        {
            name: src,
            children: [],
            size: 0,
            type: directory
        }
    ],
    size: 954,
    type: directory
}
"
`;

exports[`"Create/New" commands create package.json 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": []
}
"
`;

exports[`"Create/New" commands create source directory 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: some-random-folder-name,
            children: [],
            size: 0,
            type: directory
        }
    ],
    size: 0,
    type: directory
}
"
`;
